src/Functions/keyvaluepair/extractKeyValuePairs.cpp

```diff
@@ -0,0 +1,226 @@
+    {
+        auto offsets = ColumnUInt64::create();
+
+        auto keys = ColumnString::create();
+        auto values = ColumnString::create();
+
+        uint64_t offset = 0u;
+
+        for (auto i = 0u; i < data_column->size(); i++)
+        {
+            auto row = data_column->getDataAt(i).toView();
+
+            auto pairs_count = extractor->extract(row, keys, values);
+
+            offset += pairs_count;
+
+            offsets->insert(offset);
+        }
+
+        keys->validate();
+        values->validate();
+
+        ColumnPtr keys_ptr = std::move(keys);
+
+        return ColumnMap::create(keys_ptr, std::move(values), std::move(offsets));
+    }
+
+public:
+    ExtractKeyValuePairs() = default;
+
+    static constexpr auto name = Name::name;
+
+    String getName() const override
+    {
+        return name;
+    }
+
+    static FunctionPtr create(ContextPtr)
+    {
+        return std::make_shared<ExtractKeyValuePairs>();
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t) const override
+    {
+        auto parsed_arguments = ArgumentExtractor::extract(arguments);
+
+        auto extractor = getExtractor(parsed_arguments);
+
+        return extract(parsed_arguments.data_column, extractor);
+    }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes &) const override
+    {
+        return std::make_shared<DataTypeMap>(std::make_shared<DataTypeString>(), std::make_shared<DataTypeString>());
+    }
+
+    bool isVariadic() const override
+    {
+        return true;
+    }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo &) const override
+    {
+        return false;
+    }
+
+    std::size_t getNumberOfArguments() const override
+    {
+        return 0u;
+    }
+
+    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override
+    {
+        return {1, 2, 3, 4};
+    }
+};
+
+struct NameExtractKeyValuePairs
+{
+    static constexpr auto name = "extractKeyValuePairs";
+};
+
+struct NameExtractKeyValuePairsWithEscaping
+{
+    static constexpr auto name = "extractKeyValuePairsWithEscaping";
+};
+
+REGISTER_FUNCTION(ExtractKeyValuePairs)
+{
+    factory.registerFunction<ExtractKeyValuePairs<NameExtractKeyValuePairs, false>>(
+        Documentation(
+            R"(Extracts key-value pairs from any string. The string does not need to be 100% structured in a key value pair format;
+
+            It can contain noise (e.g. log files). The key-value pair format to be interpreted should be specified via function arguments.
+
+            A key-value pair consists of a key followed by a `key_value_delimiter` and a value. Quoted keys and values are also supported. Key value pairs must be separated by pair delimiters.
+
+            **Syntax**
+            ``` sql
+            extractKeyValuePairs(data, [key_value_delimiter], [pair_delimiter], [quoting_character])
```

Comment:
We should also decide how to unify it with another PR with `str_to_map`. Maybe `str_to_map` (or `mapFromString` can be just an alias to exctractKeyValuePair). Then the question is about arguments.

E.g., in https://docs.databricks.com/sql/language-manual/functions/str_to_map.html there're `pairDelim` in 2nd position and `keyValueDelim` in 3rd. Another similar function is https://github.com/GoogleCloudPlatform/bigquery-utils/tree/master/udfs/community#cw_map_parsem-string-pd-string-kvd-string

Should we also do like this? Do you have any opinion on the order of arguments?

Reply:
In [DataDog's parseKeyValue](https://docs.datadoghq.com/logs/log_configuration/parsing/?tab=matchers#key-value-or-logfmt) function, key value delimiter is the 2nd. That seems to be a very minor thing for me and would require re-ordering all parameters in code and tests.

Reply:
But if you think it's the best order, I can do it. No problem :)